engines/director/castmember.cpp

```diff
@@ -986,8 +986,16 @@ Common::Array<Channel> *FilmLoopCastMember::getSubChannels(Common::Rect &bbox, C
 		chan._height = height;
 
 		_subchannels.push_back(chan);
-		_subchannels[_subchannels.size() - 1].replaceWidget();
+
+	}
+	// Initialise the widgets on all of the subchannels.
+	// This has to be done once the list has been constructed, otherwise
+	// the list grow operation will erase the widgets as they aren't
+	// part of the Channel assignment constructor.
+	for (Common::Array<Channel>::iterator iter = _subchannels.begin(); iter != _subchannels.end(); ++iter) {
```

Comment:
I recommend writing: `for (auto &iter : _subchannels)`

Reply:
Oh my word that is so much nicer! Glad to be rid of the C++ iterator dance.